incubator/hnc/api/v1alpha2/hnc_config_conversion.go

```diff
@@ -0,0 +1,190 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+
+	"sigs.k8s.io/multi-tenancy/incubator/hnc/api/v1alpha1"
+)
+
+// ConvertTo converts this hncconfig to the Hub version (v1alpha1).
+func (src *HNCConfiguration) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1alpha1.HNCConfiguration)
+
+	// Spec
+	sspts := src.Spec.Types
+	dspts := []v1alpha1.TypeSynchronizationSpec{}
+	for _, sspt := range sspts {
+		dspt := v1alpha1.TypeSynchronizationSpec{}
+		dspt.APIVersion = sspt.APIVersion
+		dspt.Kind = sspt.Kind
+		switch sspt.Mode {
+		case Propagate:
+			dspt.Mode = v1alpha1.Propagate
+		case Ignore:
+			dspt.Mode = v1alpha1.Ignore
+		case Remove:
+			dspt.Mode = v1alpha1.Remove
```

Comment:
This might be more clearly expressed as a map, and lets you skip the second switch statements.

```
modeMap := map[SynchronizationMode]v1a1.SynchronizationMode{
  Propagate: v1a1.Propagate,
  Ignore: v1a1.Ignore,
  Remove: v1a1.Remove,
}
...
mapped, ok := modeMap[sspt.Mode]
if !ok {
  mapped = v1a1.SynchronizationMode(unknownModePrefix+string(sspt.Mode))
}
```

Reply:
Good suggestion updated!